### ISSUE TITLE
HTTPS Request with TLSv1.3 not working

### ISSUE THREAD

**Issue by MystixCode** [godotengine/godot#42857]
**Godot version:** 4.0

**OS including version:** Ubuntu 20.04.1 LTS (64-bit)

**Issue description:**
If I do a POST request to a webserver thats supports only TLSv1.3 with following two CipherSuites: `TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256`

I get this error: `_do_handshake: TLS handshake error: -30592`

If I change the webserver config to TLSv1.2 with common CipherSuites it works

**Comment by Calinou** [comment]
cc @Faless

**Comment by Faless** [comment]
TLS 1.3 is not yet supported by mbedTLS upstream.
When implemented, it will be supported in Godot as soon as we update the mbedTLS version accordingly:
https://tls.mbed.org/tech-updates/blog/working-towards-mbed-tls-3
ARMmbed/mbedtls#508

**Comment by MystixCode** [comment]
Nice to know. ty guys

**Comment by akien-mga** [comment]
Should we mention this in the relevant documentation?

**Comment by akien-mga** [comment]
I think we can close this as this doesn't depend on us, and if I understand correctly it will work out of the box once mbedtls gets support. This is now documented thanks to #42870.

**Comment by 4rombusiness** [comment]
I still get `_do_handshake: TLS handshake error: -30592` on Godot 4.0.3 on client side when I'm trying to connect to my server behind cloudflare.

server side codes:
```
func createServer():
	multiplayer.multiplayer_peer = null
	
	var server_certs = load("res://certs/server.crt")
	var server_key = load("res://certs/server.key")
	var server_tls_options = TLSOptions.server(server_certs, server_key)
	peer.create_server(DEF_PORT,"*",server_tls_options)
	multiplayer.multiplayer_peer = peer
	multiplayer.peer_connected.connect(self._peer_connected)
```
client side:
```
func _on_cli_pressed():
	multiplayer.multiplayer_peer = null
	var client_tls_options = TLSOptions.client_unsafe()
	peer.create_client("wss://XXX.XXX.XXX:2096" ,client_tls_options)
```

**Comment by 4rombusiness** [comment]
Ok it took 2 days till I found out the **first** parameter in `TLSOptions.server()` should be the key and the **second** one should be the cert.

It was not my mistake cuz I followed the example provided by docs:
<img width="792" alt="image" src="https://github.com/godotengine/godot/assets/74930940/212ca0c9-8a18-40ab-8831-ba999133207e">
 
And due to using `load()` without casting editor didn't show any error.

**Comment by akien-mga** [comment]
This was fixed by #77924.

In the future, please open a dedicated issue instead of commenting on a 3 year old one, as debugging your issue is unrelated to the issue which was fixed 3 years ago.
